daemons/controld/controld_cib.c

```diff
@@ -783,6 +783,55 @@ should_preserve_lock(lrmd_event_data_t *op)
     return false;
 }
 
+/*!
+ * \internal
+ * \brief Request a CIB update
+ *
+ * \param[in]     section   Section of CIB to update
+ * \param[in,out] data      New XML of CIB section to update
+ * \param[in]     options   CIB call options
+ * \param[in]     callback  If not NULL, set this as the operation callback
+ *
+ * \return Standard Pacemaker return code
+ */
+int
+controld_update_cib(const char *section, xmlNode *data, int options,
+                    void (*callback)(xmlNode *, int, int, xmlNode *, void *))
+{
+    int cib_rc = -ENOTCONN;
+
+    CRM_ASSERT(data != NULL);
+
+    if (controld_globals.cib_conn != NULL) {
+        cib_rc = cib_internal_op(controld_globals.cib_conn,
+                                 PCMK__CIB_REQUEST_MODIFY, NULL, section,
+                                 data, NULL, options, NULL);
+        if (cib_rc >= 0) {
+            crm_debug("Submitted CIB update %d for %s section",
+                      cib_rc, section);
+        }
+    }
+
+    if (callback == NULL) {
+        if (cib_rc < 0) {
+            crm_err("Failed to update CIB %s section: %s",
+                    section, pcmk_rc_str(pcmk_legacy2rc(cib_rc)));
+        }
+
+    } else {
+        if ((cib_rc >= 0) && (callback == cib_rsc_callback)) {
```

Comment:
Yeah, the functions should never return 0 for async ops, but if they did, it should be a valid call ID. So really, either check is justifiable, but since we always use < 0 to check for errors, it makes sense to use >= 0 for success. Of course it will be moot when everything eventually uses standard codes :)